app/src/main/java/org/dwoc/khet/gameControl/Piece.kt

```diff
@@ -0,0 +1,31 @@
+package org.dwoc.khet.gameControl
+
+import org.dwoc.khet.models.enums.Direction
+import org.dwoc.khet.models.enums.Player
+
+// Each Types of Piece will Extend this Piece Class
+abstract class Piece(val PlayerId : Player, var PosX : Int, var PosY : Int, var Orientation : Direction)
+{
+
+  //Function MovePiece will set the PosX and PosY of the Piece
+  fun MovePiece (PosX : Int,PosY : Int)
+  {
+    this.PosX = PosX
+    this.PosY = PosY
+  }
+
+  //Function RotatePiece will Rotate the Piece's Orientation by the Said Amount
+  fun RotatePiece (amount : Int)
+  {
+     var value: Int = amount + Orientation.ordinal + 4
+    value %= 4
```

Comment:
account for negative as well. if it becomes -1 set it to 3

Reply:
The + 4 in the end will make it positive then we will be able to use the %4 to get the remainder!

Reply:
The + 4 in line 20 after the Orientation.ordinal will account the negative 1 and makes it positive then the %4 will give values in range 0 to 3